Source/SwiftLintFramework/Rules/Performance/ContainsOverRangeNotNilRule.swift

```diff
@@ -0,0 +1,34 @@
+import SourceKittenFramework
+
+public struct ContainsOverRangeNotNilRule: CallPairRule, OptInRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "contains_over_range_not_nil",
+        name: "Contains over range not nil",
+        description: "Prefer `contains` over `range(of:) != nil`.",
+        kind: .performance,
+        nonTriggeringExamples: [
+            "let range = myString.range(of: \"Test\")",
+            "myString.contains(\"Test\")",
+            "!myString.contains(\"Test\")",
+            "resourceString.range(of: rule.regex, options: .regularExpression) != nil"
+        ],
+        triggeringExamples: ["!=", "=="].flatMap { comparison in
+            return [
+                "↓myString.range(of: \"Test\") \(comparison) nil"
+            ]
+        }
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "\\)\\s*(==|!=)\\s*nil"
+        return validate(file: file, pattern: pattern, patternSyntaxKinds: [.keyword],
+                        callNameSuffix: ".range", severity: configuration.severity,
+                        reason: "Prefer `contains` over `range(of:) != nil`") { expression in
+                            return expression.enclosedArguments.count == 1
```

Comment:
on a second thought, can we compare the param name too to be extra careful?

This should do the trick: 

```swift
return expression.enclosedArguments.map { $0.name } == ["of"]
```

Reply:
I'll update with the rename and the parameter check improvement.

Reply:
Done